ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/transactions/PendingTransactionsTest.java

```diff
@@ -409,4 +423,46 @@ private Transaction createTransaction(final int transactionNumber) {
         .nonce(transactionNumber)
         .createTransaction(KEYS1);
   }
+
+  @Test
+  public void shouldEvictSingleOldTransaction() {
+    final long TRANSACTION_EVICTION_INTERVAL_MS = TimeUnit.SECONDS.toMillis(1);
+    final PendingTransactions transactions =
+        new PendingTransactions(
+            timerUtil,
+            TRANSACTION_EVICTION_INTERVAL_MS,
+            MAX_TRANSACTIONS,
+            TestClock.fixed(),
+            metricsSystem);
+    transactions.addRemoteTransaction(transaction1);
+    assertThat(transactions.size()).isEqualTo(1);
+    try {
+      TimeUnit.SECONDS.sleep(2);
+    } catch (Exception ignored) {
+    }
+    assertThat(transactions.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldEvictMultipleOldTransactions() {
+    final long TRANSACTION_EVICTION_INTERVAL_MS = TimeUnit.SECONDS.toMillis(1);
+    final PendingTransactions transactions =
+        new PendingTransactions(
+            timerUtil,
+            TRANSACTION_EVICTION_INTERVAL_MS,
+            MAX_TRANSACTIONS,
+            TestClock.fixed(),
+            metricsSystem);
+
+    transactions.addRemoteTransaction(transaction1);
+    assertThat(transactions.size()).isEqualTo(1);
+    transactions.addRemoteTransaction(transaction2);
+    assertThat(transactions.size()).isEqualTo(2);
+
+    try {
+      TimeUnit.SECONDS.sleep(2);
+    } catch (Exception ignored) {
+    }
+    assertThat(transactions.size()).isEqualTo(0);
```

Comment:
We should have a test that has a combination of old and new transactions so we know it doesn't just evict all transactions.